commit 8414fe11ba2ff2422c4b73bbef3b035a2562c88a
Author: Tobin C. Harding <me@tobin.cc>
Date:   Wed Feb 15 14:27:23 2017 +1100

    staging: fbtft: Fix buffer overflow vulnerability
    
    Module copies a user supplied string (module parameter) into a buffer
    using strncpy() and does not check that the buffer is null terminated.
    
    Replace call to strncpy() with call to strlcpy() ensuring that the
    buffer is null terminated.
    
    Signed-off-by: Tobin C. Harding <me@tobin.cc>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/drivers/staging/fbtft/fbtft_device.c b/drivers/staging/fbtft/fbtft_device.c
index 9ffb9cecc465..6b9e8adda984 100644
--- a/drivers/staging/fbtft/fbtft_device.c
+++ b/drivers/staging/fbtft/fbtft_device.c
@@ -1483,7 +1483,7 @@ static int __init fbtft_device_init(void)
 			displays[i].pdev->name = name;
 			displays[i].spi = NULL;
 		} else {
-			strncpy(displays[i].spi->modalias, name, SPI_NAME_SIZE);
+			strlcpy(displays[i].spi->modalias, name, SPI_NAME_SIZE);
 			displays[i].pdev = NULL;
 		}
 	}